DicomRtImportExport/Logic/vtkSlicerDicomRtReader.cxx

```diff
@@ -77,18 +77,53 @@ class vtkSlicerDicomRtReader::vtkInternal
   public:
     BeamLimitingDeviceEntry()
       :
-      Distance(-1.),
-      NumberOfLeafJawPairs(0)
+      SourceIsoDistance(400.),
+      NumberOfPairs(0)
     {
     }
-    /// Source to BeamLimitingDevice (MLC) distance for RTPlan
+    /// Source to BeamLimitingDevice (MLC) distance for RTPlan (first element)
     /// or Isocenter to BeamLimitingDevice (MLC) distance for RTIonPlan
-    double Distance; // loaded, but not used
-    unsigned int NumberOfLeafJawPairs;
+    double SourceIsoDistance;
+    unsigned int NumberOfPairs;
     /// MLC position boundaries: Raw DICOM values
     std::vector<double> LeafPositionBoundary;
   };
 
+  /// Structure storing a beam range shifter parameters associated with ion beam
+  class RangeShifterEntry
+  {
+  public:
+    RangeShifterEntry()
+      :
+      Number(0),
+      Type("BINARY")
+    {
+    }
+    int Number;
+    std::string ID;
+    std::string AccessoryCode;
+    std::string Type; // "ANALOG", "BINARY"
+    std::string Description;
+  };
+
+  /// Structure storing a treatment compensator parameters associated with beam
+  class CompensatorEntry

```

Comment:
Is this class and BlockEntry future work? Since they don't contain anything... In that case please add "//TODO: ..." comments

Reply:
I will do this.